core/harambe_core/parser/type_price.py

```diff
@@ -0,0 +1,82 @@
+import re
+from typing import Any, Optional, Union
+
+from price_parser import Price
+from pydantic import BeforeValidator
+from typing_extensions import Annotated, Literal
+
+from harambe_core.parser.constants import CURRENCY_MAP, PRICE_NOT_AVAILABLE_PHRASES
+from harambe_core.parser.type_currency import ParserTypeCurrency
+
+# this will capture only the amount
+number_pattern = re.compile(r"-?.?\d[\d,-\.]*")
+
+
+class ParserTypePrice:
+    def __new__(cls, required: bool = True) -> Any:
+        validator = BeforeValidator(cls.validate_price)
+        base = dict[
+            Literal["currency", "raw_currency", "amount", "raw_price"],
+            Optional[Union[str, float]],
+        ]
```

Comment:
refactor out as a type so it can be shared on line 32

Reply:
also instead of: dict[
            Literal["currency", "raw_currency", "amount", "raw_price"],
            Optional[Union[str, float]],
        ]
        
     why not use a TypedDict?